File: Src/ButterflyNet.Parser/Scopes/LineBreakScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class LineBreakScope : InlineScope {

		public override bool CanNestText { get { return false; } }

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenLineBreak();
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseLineBreak();
		}
	}
}

File: src/js/src/scopes/LineBreakScope.js
function LineBreakScope() {
	this.type = ScopeTypeCache.lineBreak;
	this.level = ScopeLevel.inline;
	
	this.open = function(analyzer) {
		analyzer.openLineBreak();
	};
	this.close = function(analyzer) {
		analyzer.closeLineBreak();
	};
}

extend(Scope, LineBreakScope);